flocker/acceptance/endtoend/test_installer.py

```diff
@@ -0,0 +1,319 @@
+    cluster_cert = certificates_path.child(b"cluster.crt")
+    user_cert = certificates_path.child(b"user1.crt")
+    user_key = certificates_path.child(b"user1.key")
+    user_credential = UserCredential.from_files(user_cert, user_key)
+    cluster = Cluster(
+        control_node=ControlService(
+            public_address=test_case.control_node_ip.encode("ascii")),
+        nodes=[],
+        treq=treq_with_authentication(
+            reactor, cluster_cert, user_cert, user_key),
+        client=FlockerClient(
+            reactor, test_case.control_node_ip.encode("ascii"),
+            REST_API_PORT, cluster_cert, user_cert, user_key
+        ),
+        certificates_path=certificates_path,
+        cluster_uuid=user_credential.cluster_uuid,
+    )
+    d_node1_compose = remote_docker_compose(test_case.client_ip,
+                                            test_case.docker_host,
+                                            COMPOSE_NODE0, 'stop')
+    d_node1_compose.addCallback(
+        lambda ignored: remote_docker_compose(
+            test_case.client_ip,
+            test_case.docker_host,
+            COMPOSE_NODE0, 'rm', '-f'
+        )
+    )
+
+    d_node2_compose = remote_docker_compose(
+        test_case.client_ip, test_case.docker_host, COMPOSE_NODE1, 'stop')
+    d_node2_compose.addCallback(
+        lambda ignored: remote_docker_compose(
+            test_case.client_ip, test_case.docker_host, COMPOSE_NODE1,
+            'rm', '-f')
+    )
+
+    d = gather_deferreds([d_node1_compose, d_node2_compose])
+    d.addCallback(
+        lambda ignored: cluster.clean_nodes()
+    )
+
+    d.addCallback(
+        lambda ignored: delete_cloudformation_stack(test_case.stack_id)
+    )
+
+    return d
+
+
+def get_stack_report(stack_id):
+    output = check_output(
+        ['aws', '--region', REGION, 'cloudformation', 'describe-stacks',
+         '--stack-name', stack_id]
+    )
+    results = json.loads(output)
+    return results['Stacks'][0]
+
+
+def wait_for_stack_status(stack_id, target_status, time_limit=600):
+    start_time = time.time()
+    while True:
+        stack_report = get_stack_report(stack_id)
+        current_status = stack_report['StackStatus']
+        if current_status == target_status:
+            return stack_report
+        time_running = time.time() - start_time
+        if time_running > time_limit:
+            Message.new(
+                message='Timeout waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+            ).write()
+            return False
+        else:
+            Message.new(
+                message='Waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+                time_running=time_running,
+            ).write()
+
+            time.sleep(10)
```

Comment:
Please use on of the helpers in `flocker.common.retry`. Feel free to tweak them if they aren't directly suitable.